packages/rest/src/__tests__/integration/rest.application.integration.ts

```diff
@@ -98,7 +98,9 @@ describe('RestApplication (integration)', () => {
     restApp.static('/', ASSETS);
     await restApp.start();
     client = createRestAppClient(restApp);
-    await client.get('/html/index.html').expect(200);
+    // No more `/html` in client as `restApp.restServer.url` already contains
+    // the base path `/html`
```

Comment:
This change will make it more difficult to use `createRestAppClient` for testing endpoints that are ignoring basePath, e.g. `/openapi.json`.

I think this is also a breaking change for consumers of `createRestAppClient`.

Personally, I'd prefer to preserve the current behavior of `reateRestAppClient` where it creates supertest instance for the host+port only.

Maybe it's a sign that we need two server-level properties?
 - protocol+host+port only (e.g. `http://127.0.0.1:3000`)
 - base URL (e.g. `http://127.0.0.1:3000/my-base-path`)

Let's discuss.

Reply:
I introduced `rootUrl` for the protocol + host + port and changed `createRestAppClient` to use `rootUrl` to maintain the backward compatibility.

I use `url` with base path to be consistent with OpenAPI spec server url.